### PR TITLE
Add metadata overflow menu (new branch)

### PR DESCRIPTION
**Pull request by sxian0613**
add overflow menu for meta data （new branch)

1. Removed the meta data button and replaced it with an overflow menu.
2. Added "Bounding Box" and "Attributes" with two checkboxes.
3. Added "confidence" and "speed" to the Attributes showing on the bounding boxes.
4. Controlled the showing of bounding boxes and Attributes by checking the checkboxes.
5. Cleaned the code.

### CODE REVIEW

File: packages/web-components/src/player-component/UI/ui.class.ts [hunk 1]
```diff
@@ -201,6 +201,63 @@ export class OverflowMenu extends shaka.ui.OverflowMenu {
...
+                       private showBoxCallBack: (isOn: boolean) => void, 
+                       private showAttributesCallBack: (isOn: boolean) => void) {
+        super(parent, controls);
+        this.init();
+    }
+
+    public createChildren_() {
+        const showBoundingBox = document.createElement('div');
+        showBoundingBox.innerText = ' Bounding Box ';
+        const BoxCheckbox = document.createElement('input');
```
Comment (giakas) [inline packages/web-components/src/player-component/UI/ui.class.ts:221]: Why are we not using Fast controls here?

**Review by giakas — CHANGES_REQUESTED** [review]

File: packages/web-components/src/player-component/UI/ui.class.ts (hunk 1 above)
Comment (sxian0613) [reply packages/web-components/src/player-component/UI/ui.class.ts]: We are now using FAST checkbox.

File: global.export.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ FASTButton;
 FASTMenu;
 FASTMenuItem;
 FASTSlider;
+FASTCheckbox;
```
Comment (giakas) [inline global.export.ts:12]: >FASTCheckbox [](http://example.com/codeflow?start=0&length=12)

dont you need to do this for npm exports file too?

File: packages/web-components/src/player-component/UI/ui.class.ts [hunk 3]
```diff
@@ -201,6 +201,61 @@ export class OverflowMenu extends shaka.ui.OverflowMenu {
...
+                       controls: any, 
+                       private showBoxCallBack: (isOn: boolean) => void, 
+                       private showAttributesCallBack: (isOn: boolean) => void) {
+        super(parent, controls);
+        this.init();
+    }
+
+    public createChildren_() {
+        const showBoundingBox = document.createElement('div');
+        showBoundingBox.innerText = ' Bounding Box ';
```
Comment (giakas) [inline packages/web-components/src/player-component/UI/ui.class.ts:220]: > Bounding Box [](http://example.com/codeflow?start=37&length=13)

you should be getting this strings from localization.

File: packages/web-components/src/player-component/UI/ui.class.ts [hunk 4]
```diff
@@ -201,6 +201,61 @@ export class OverflowMenu extends shaka.ui.OverflowMenu {
...
+        const BoxCheckbox = document.createElement('fast-checkbox');
+        this.eventManager.listen(BoxCheckbox, 'click', () => {
+            this.isBoxOn = !this.isBoxOn;
+            this.showBoxCallBack(this.isBoxOn);
+        });
+        showBoundingBox.appendChild(BoxCheckbox);
+        this.overflowMenu_.appendChild(showBoundingBox);
+
+        const showAtrribute = document.createElement('div');
+        showAtrribute.innerText = ' Attributes ';
```
Comment (giakas) [inline packages/web-components/src/player-component/UI/ui.class.ts:230]: >Attributes [](http://example.com/codeflow?start=36&length=10)

same here.. from localizatoin.. add it to en.json and de.json and use it from localization values

File: packages/web-components/src/player-component/UI/ui.class.ts [hunk 5]
```diff
@@ -201,6 +201,61 @@ export class OverflowMenu extends shaka.ui.OverflowMenu {
...
+        this.eventManager.listen(AttributesCheckbox, 'click', () => {
+            this.showAttributesCallBack(this.isAttributesOn)
+        });
+        showAtrribute.appendChild(AttributesCheckbox);
+        this.overflowMenu_.appendChild(showAtrribute);
+
+        const settingsLabel = document.createElement('label');
+        settingsLabel.classList.add('settings-header');
+        const settingsSpan = document.createElement('span');
+        settingsSpan.innerText = 'Meta Data Settings';
```
Comment (giakas) [inline packages/web-components/src/player-component/UI/ui.class.ts:241]: >Meta [](http://example.com/codeflow?start=34&length=4)

same here

File: packages/web-components/src/player-component/UI/ui.class.ts [hunk 6]
```diff
@@ -201,6 +201,61 @@ export class OverflowMenu extends shaka.ui.OverflowMenu {
...
+            this.showBoxCallBack(this.isBoxOn);
+        });
+        showBoundingBox.appendChild(BoxCheckbox);
+        this.overflowMenu_.appendChild(showBoundingBox);
+
+        const showAtrribute = document.createElement('div');
+        showAtrribute.innerText = ' Attributes ';
+        const AttributesCheckbox = document.createElement('fast-checkbox');
+        this.eventManager.listen(AttributesCheckbox, 'click', () => {
+            this.showAttributesCallBack(this.isAttributesOn)
```
Comment (giakas) [inline packages/web-components/src/player-component/UI/ui.class.ts:233]: >this.isAttributesOn [](http://example.com/codeflow?start=40&length=19)

you don't need to reverse this flag. like you did on 223?

**Pull request closed without merging**